lib/github-provider.js

```diff
@@ -68,9 +71,57 @@ function mapGithubCommitsSince(camp, githubApiUrl, githubAuth) {
     }));
 }
 
+function mapGithubReleaseDate(camp, githubApiUrl, githubAuth) {
+  camp.route(/^\/github\/release-date\/([^\/]+)\/([^\/]+)\.(svg|png|gif|jpg|json)$/,
+  cache(function(data, match, sendBadge, request) {
+    const user = match[1];  // eg, microsoft
+    const repo = match[2];  // eg, vscode
+    const format = match[3];
+    const apiUrl = `${githubApiUrl}/repos/${user}/${repo}/releases/latest`;
+    const badgeData = getBadgeData('release date', data);
+    if (badgeData.template === 'social') {
+      badgeData.logo = getLogo('github', data);
+    }
+    githubAuth.request(request, apiUrl, {}, function(err, res, buffer) {
+      if (err != null) {
+        badgeData.text[1] = 'inaccessible';
+        sendBadge(format, badgeData);
+        return;
+      }
+      try {
+        const data = JSON.parse(buffer);
+  
+        if (!data.created_at) throw 'Project has no releases';
```

Comment:
Instead of throwing, this should return a badge with a helpful message.

Reply:
actually, problem is that `data` is always a object even if the the repo is the exist. So,   `data.created_at` will not throw any error as it will be always null or a valid value. 

If you think, we should return a valid message, then I've to check the status code first then `data.created_at`. -- NO PROBLEM AT ALL.

Should i do this?

Reply:
Ah, I gotcha. Yes, it'd be better to add a test for a nonexistent repo, determine how to detect that condition, and return a useful message.

Reply:
ok! Then, I should check the returned status code